### ISSUE TITLE
Tests failures under Python 3.8

### ISSUE THREAD

**Issue by eclipseo** [pysal/libpysal#177]
Python 3.8beta3, Release 4.1.0

Environment:
```
DEBUG util.py:587:   python3-devel                     i686   3.8.0~b3-4.fc32
DEBUG util.py:587:   python3-geomet                    noarch 0.2.1-1.fc32
DEBUG util.py:587:   python3-geopandas                 noarch 0.6.0-0.1.rc1.fc32
DEBUG util.py:587:   python3-matplotlib                i686   3.1.1-1.fc32.1
DEBUG util.py:587:   python3-networkx                  noarch 2.3-4.fc32
DEBUG util.py:587:   python3-nose                      noarch 1.3.7-25.fc32
DEBUG util.py:587:   python3-nose-exclude              noarch 0.5.0-6.fc32
DEBUG util.py:587:   python3-nose-progressive          noarch 1.5.1-18.fc32
DEBUG util.py:587:   python3-numpy                     i686   1:1.17.0-3.fc32
DEBUG util.py:587:   python3-numpydoc                  noarch 0.9.1-3.fc32
DEBUG util.py:587:   python3-pandas                    i686   0.25.1-1.fc32
DEBUG util.py:587:   python3-rtree                     noarch 0.8.3-11.fc32
DEBUG util.py:587:   python3-scipy                     i686   1.2.1-7.fc32
DEBUG util.py:587:   python3-setuptools                noarch 41.0.1-8.fc32
DEBUG util.py:587:   python3-sphinx                    noarch 1:2.1.2-5.fc32
DEBUG util.py:587:   python3-sphinx-bootstrap-theme    noarch 0.4.13-13.fc32
DEBUG util.py:587:   python3-sphinxcontrib-bibtex      noarch 1.0.0~a0.20170423git5fa80ec5-3.fc32
DEBUG util.py:587:   python3-sqlalchemy                i686   1.3.6-2.fc32
```

```
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_alpha_shapes (libpysal.cg.tests.test_ashapes.Test_Alpha_Shapes)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/tests/test_ashapes.py", line 25, in setUp
BUILDSTDERR:     self.a05 = geopandas.read_file(os.path.join(this_directory, 'data/alpha_05.shp')).geometry.item()
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/base.py", line 740, in item
BUILDSTDERR:     return self.values.item()
BUILDSTDERR: AttributeError: 'GeometryArray' object has no attribute 'item'
BUILDSTDERR: -------------------- >> begin captured logging << --------------------
BUILDSTDERR: fiona.env: DEBUG: Entering env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Starting outermost env
BUILDSTDERR: fiona.env: DEBUG: No GDAL environment exists
BUILDSTDERR: fiona.env: DEBUG: New GDAL environment <fiona._env.GDALEnv object at 0xe72d3280> created
BUILDSTDERR: fiona._env: DEBUG: Logging error handler pushed.
BUILDSTDERR: fiona._env: DEBUG: All drivers registered.
BUILDSTDERR: fiona._env: DEBUG: GDAL data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: PROJ data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: Started GDALEnv <fiona._env.GDALEnv object at 0xe72d3280>.
BUILDSTDERR: fiona.env: DEBUG: Updated existing <fiona._env.GDALEnv object at 0xe72d3280> with options {}
BUILDSTDERR: fiona.env: DEBUG: Entered env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Index: 0
BUILDSTDERR: fiona.collection: DEBUG: Flushed buffer
BUILDSTDERR: fiona.collection: DEBUG: Stopped session
BUILDSTDERR: fiona.env: DEBUG: Exiting env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Cleared existing <fiona._env.GDALEnv object at 0xe72d3280> options
BUILDSTDERR: fiona._env: DEBUG: Stopping GDALEnv <fiona._env.GDALEnv object at 0xe72d3280>.
BUILDSTDERR: fiona._env: DEBUG: Error handler popped.
BUILDSTDERR: fiona._env: DEBUG: Stopped GDALEnv <fiona._env.GDALEnv object at 0xe72d3280>.
BUILDSTDERR: fiona.env: DEBUG: Exiting outermost env
BUILDSTDERR: fiona.env: DEBUG: Exited env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Entering env context: <fiona.env.Env object at 0xe73fbfe8>
BUILDSTDERR: fiona.env: DEBUG: Starting outermost env
BUILDSTDERR: fiona.env: DEBUG: No GDAL environment exists
BUILDSTDERR: fiona.env: DEBUG: New GDAL environment <fiona._env.GDALEnv object at 0xe72d3388> created
BUILDSTDERR: fiona._env: DEBUG: Logging error handler pushed.
BUILDSTDERR: fiona._env: DEBUG: All drivers registered.
BUILDSTDERR: fiona._env: DEBUG: GDAL data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: PROJ data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: Started GDALEnv <fiona._env.GDALEnv object at 0xe72d3388>.
BUILDSTDERR: fiona.env: DEBUG: Updated existing <fiona._env.GDALEnv object at 0xe72d3388> with options {}
BUILDSTDERR: fiona.env: DEBUG: Entered env context: <fiona.env.Env object at 0xe73fbfe8>
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Index: 0
BUILDSTDERR: fiona.collection: DEBUG: Flushed buffer
BUILDSTDERR: fiona.collection: DEBUG: Stopped session
BUILDSTDERR: fiona.env: DEBUG: Exiting env context: <fiona.env.Env object at 0xe73fbfe8>
BUILDSTDERR: fiona.env: DEBUG: Cleared existing <fiona._env.GDALEnv object at 0xe72d3388> options
BUILDSTDERR: fiona._env: DEBUG: Stopping GDALEnv <fiona._env.GDALEnv object at 0xe72d3388>.
BUILDSTDERR: fiona._env: DEBUG: Error handler popped.
BUILDSTDERR: fiona._env: DEBUG: Stopped GDALEnv <fiona._env.GDALEnv object at 0xe72d3388>.
BUILDSTDERR: fiona.env: DEBUG: Exiting outermost env
BUILDSTDERR: fiona.env: DEBUG: Exited env context: <fiona.env.Env object at 0xe73fbfe8>
BUILDSTDERR: --------------------- >> end captured logging << ---------------------
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_auto (libpysal.cg.tests.test_ashapes.Test_Alpha_Shapes)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/tests/test_ashapes.py", line 25, in setUp
BUILDSTDERR:     self.a05 = geopandas.read_file(os.path.join(this_directory, 'data/alpha_05.shp')).geometry.item()
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/base.py", line 740, in item
BUILDSTDERR:     return self.values.item()
BUILDSTDERR: AttributeError: 'GeometryArray' object has no attribute 'item'
BUILDSTDERR: -------------------- >> begin captured logging << --------------------
BUILDSTDERR: fiona.env: DEBUG: Entering env context: <fiona.env.Env object at 0xe751ebf8>
BUILDSTDERR: fiona.env: DEBUG: Starting outermost env
BUILDSTDERR: fiona.env: DEBUG: No GDAL environment exists
BUILDSTDERR: fiona.env: DEBUG: New GDAL environment <fiona._env.GDALEnv object at 0xe72d3aa8> created
BUILDSTDERR: fiona._env: DEBUG: Logging error handler pushed.
BUILDSTDERR: fiona._env: DEBUG: All drivers registered.
BUILDSTDERR: fiona._env: DEBUG: GDAL data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: PROJ data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: Started GDALEnv <fiona._env.GDALEnv object at 0xe72d3aa8>.
BUILDSTDERR: fiona.env: DEBUG: Updated existing <fiona._env.GDALEnv object at 0xe72d3aa8> with options {}
BUILDSTDERR: fiona.env: DEBUG: Entered env context: <fiona.env.Env object at 0xe751ebf8>
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Index: 0
BUILDSTDERR: fiona.collection: DEBUG: Flushed buffer
BUILDSTDERR: fiona.collection: DEBUG: Stopped session
BUILDSTDERR: fiona.env: DEBUG: Exiting env context: <fiona.env.Env object at 0xe751ebf8>
BUILDSTDERR: fiona.env: DEBUG: Cleared existing <fiona._env.GDALEnv object at 0xe72d3aa8> options
BUILDSTDERR: fiona._env: DEBUG: Stopping GDALEnv <fiona._env.GDALEnv object at 0xe72d3aa8>.
BUILDSTDERR: fiona._env: DEBUG: Error handler popped.
BUILDSTDERR: fiona._env: DEBUG: Stopped GDALEnv <fiona._env.GDALEnv object at 0xe72d3aa8>.
BUILDSTDERR: fiona.env: DEBUG: Exiting outermost env
BUILDSTDERR: fiona.env: DEBUG: Exited env context: <fiona.env.Env object at 0xe751ebf8>
BUILDSTDERR: fiona.env: DEBUG: Entering env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Starting outermost env
BUILDSTDERR: fiona.env: DEBUG: No GDAL environment exists
BUILDSTDERR: fiona.env: DEBUG: New GDAL environment <fiona._env.GDALEnv object at 0xe72d3988> created
BUILDSTDERR: fiona._env: DEBUG: Logging error handler pushed.
BUILDSTDERR: fiona._env: DEBUG: All drivers registered.
BUILDSTDERR: fiona._env: DEBUG: GDAL data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: PROJ data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: Started GDALEnv <fiona._env.GDALEnv object at 0xe72d3988>.
BUILDSTDERR: fiona.env: DEBUG: Updated existing <fiona._env.GDALEnv object at 0xe72d3988> with options {}
BUILDSTDERR: fiona.env: DEBUG: Entered env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Index: 0
BUILDSTDERR: fiona.collection: DEBUG: Flushed buffer
BUILDSTDERR: fiona.collection: DEBUG: Stopped session
BUILDSTDERR: fiona.env: DEBUG: Exiting env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Cleared existing <fiona._env.GDALEnv object at 0xe72d3988> options
BUILDSTDERR: fiona._env: DEBUG: Stopping GDALEnv <fiona._env.GDALEnv object at 0xe72d3988>.
BUILDSTDERR: fiona._env: DEBUG: Error handler popped.
BUILDSTDERR: fiona._env: DEBUG: Stopped GDALEnv <fiona._env.GDALEnv object at 0xe72d3988>.
BUILDSTDERR: fiona.env: DEBUG: Exiting outermost env
BUILDSTDERR: fiona.env: DEBUG: Exited env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: --------------------- >> end captured logging << ---------------------
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_small_n (libpysal.cg.tests.test_ashapes.Test_Alpha_Shapes)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/tests/test_ashapes.py", line 25, in setUp
BUILDSTDERR:     self.a05 = geopandas.read_file(os.path.join(this_directory, 'data/alpha_05.shp')).geometry.item()
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/base.py", line 740, in item
BUILDSTDERR:     return self.values.item()
BUILDSTDERR: AttributeError: 'GeometryArray' object has no attribute 'item'
BUILDSTDERR: -------------------- >> begin captured logging << --------------------
BUILDSTDERR: fiona.env: DEBUG: Entering env context: <fiona.env.Env object at 0xe751e238>
BUILDSTDERR: fiona.env: DEBUG: Starting outermost env
BUILDSTDERR: fiona.env: DEBUG: No GDAL environment exists
BUILDSTDERR: fiona.env: DEBUG: New GDAL environment <fiona._env.GDALEnv object at 0xe72d3b68> created
BUILDSTDERR: fiona._env: DEBUG: Logging error handler pushed.
BUILDSTDERR: fiona._env: DEBUG: All drivers registered.
BUILDSTDERR: fiona._env: DEBUG: GDAL data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: PROJ data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: Started GDALEnv <fiona._env.GDALEnv object at 0xe72d3b68>.
BUILDSTDERR: fiona.env: DEBUG: Updated existing <fiona._env.GDALEnv object at 0xe72d3b68> with options {}
BUILDSTDERR: fiona.env: DEBUG: Entered env context: <fiona.env.Env object at 0xe751e238>
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Index: 0
BUILDSTDERR: fiona.collection: DEBUG: Flushed buffer
BUILDSTDERR: fiona.collection: DEBUG: Stopped session
BUILDSTDERR: fiona.env: DEBUG: Exiting env context: <fiona.env.Env object at 0xe751e238>
BUILDSTDERR: fiona.env: DEBUG: Cleared existing <fiona._env.GDALEnv object at 0xe72d3b68> options
BUILDSTDERR: fiona._env: DEBUG: Stopping GDALEnv <fiona._env.GDALEnv object at 0xe72d3b68>.
BUILDSTDERR: fiona._env: DEBUG: Error handler popped.
BUILDSTDERR: fiona._env: DEBUG: Stopped GDALEnv <fiona._env.GDALEnv object at 0xe72d3b68>.
BUILDSTDERR: fiona.env: DEBUG: Exiting outermost env
BUILDSTDERR: fiona.env: DEBUG: Exited env context: <fiona.env.Env object at 0xe751e238>
BUILDSTDERR: fiona.env: DEBUG: Entering env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Starting outermost env
BUILDSTDERR: fiona.env: DEBUG: No GDAL environment exists
BUILDSTDERR: fiona.env: DEBUG: New GDAL environment <fiona._env.GDALEnv object at 0xe72d32c8> created
BUILDSTDERR: fiona._env: DEBUG: Logging error handler pushed.
BUILDSTDERR: fiona._env: DEBUG: All drivers registered.
BUILDSTDERR: fiona._env: DEBUG: GDAL data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: PROJ data files are available at built-in paths
BUILDSTDERR: fiona._env: DEBUG: Started GDALEnv <fiona._env.GDALEnv object at 0xe72d32c8>.
BUILDSTDERR: fiona.env: DEBUG: Updated existing <fiona._env.GDALEnv object at 0xe72d32c8> with options {}
BUILDSTDERR: fiona.env: DEBUG: Entered env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Layer has no coordinate system
BUILDSTDERR: fiona.ogrext: DEBUG: Projection not found (cogr_crs was NULL)
BUILDSTDERR: fiona.ogrext: DEBUG: Index: 0
BUILDSTDERR: fiona.collection: DEBUG: Flushed buffer
BUILDSTDERR: fiona.collection: DEBUG: Stopped session
BUILDSTDERR: fiona.env: DEBUG: Exiting env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: fiona.env: DEBUG: Cleared existing <fiona._env.GDALEnv object at 0xe72d32c8> options
BUILDSTDERR: fiona._env: DEBUG: Stopping GDALEnv <fiona._env.GDALEnv object at 0xe72d32c8>.
BUILDSTDERR: fiona._env: DEBUG: Error handler popped.
BUILDSTDERR: fiona._env: DEBUG: Stopped GDALEnv <fiona._env.GDALEnv object at 0xe72d32c8>.
BUILDSTDERR: fiona.env: DEBUG: Exiting outermost env
BUILDSTDERR: fiona.env: DEBUG: Exited env context: <fiona.env.Env object at 0xe751ebe0>
BUILDSTDERR: --------------------- >> end captured logging << ---------------------
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_rtree (libpysal.cg.tests.test_rtree.Pyrtree_Tester)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/tests/test_rtree.py", line 27, in test_rtree
BUILDSTDERR:     t.insert(object, self.objects[object])
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/rtree.py", line 200, in insert
BUILDSTDERR:     self.cursor.insert(o, orect)
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/rtree.py", line 417, in insert
BUILDSTDERR:     self._balance()
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/rtree.py", line 451, in _balance
BUILDSTDERR:     t = time.clock()
BUILDSTDERR: AttributeError: module 'time' has no attribute 'clock'
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_rtree (libpysal.cg.tests.test_rtree.Pyrtree_Tester)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/tests/test_rtree.py", line 27, in test_rtree
BUILDSTDERR:     t.insert(object, self.objects[object])
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/rtree.py", line 200, in insert
BUILDSTDERR:     self.cursor.insert(o, orect)
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/rtree.py", line 417, in insert
BUILDSTDERR:     self._balance()
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/cg/rtree.py", line 451, in _balance
BUILDSTDERR:     t = time.clock()
BUILDSTDERR: AttributeError: module 'time' has no attribute 'clock'
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: test_filter (libpysal.weights.tests.test_adjlist.Test_Adjlist)
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/weights/tests/test_adjlist.py", line 32, in test_filter
BUILDSTDERR:     alist = grid.to_adjlist(remove_symmetric=True)
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/weights/weights.py", line 263, in to_adjlist
BUILDSTDERR:     return adjtools.filter_adjlist(adjlist) if remove_symmetric else adjlist
BUILDSTDERR:   File "/builddir/build/BUILD/libpysal-4.1.0/libpysal/weights/adjtools.py", line 191, in filter_adjlist
BUILDSTDERR:     to_drop = adjlist.query('focal == {} and neighbor == {}'
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/frame.py", line 3184, in query
BUILDSTDERR:     res = self.eval(expr, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/frame.py", line 3300, in eval
BUILDSTDERR:     return _eval(expr, inplace=inplace, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/eval.py", line 322, in eval
BUILDSTDERR:     parsed_expr = Expr(expr, engine=engine, parser=parser, env=env, truediv=truediv)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 827, in __init__
BUILDSTDERR:     self.terms = self.parse()
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 844, in parse
BUILDSTDERR:     return self._visitor.visit(self.expr)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 441, in visit
BUILDSTDERR:     return visitor(node, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 447, in visit_Module
BUILDSTDERR:     return self.visit(expr, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 441, in visit
BUILDSTDERR:     return visitor(node, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 450, in visit_Expr
BUILDSTDERR:     return self.visit(node.value, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 441, in visit
BUILDSTDERR:     return visitor(node, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 771, in visit_BoolOp
BUILDSTDERR:     return reduce(visitor, operands)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 764, in visitor
BUILDSTDERR:     lhs = self._try_visit_binop(x)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 760, in _try_visit_binop
BUILDSTDERR:     return self.visit(bop)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 441, in visit
BUILDSTDERR:     return visitor(node, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 744, in visit_Compare
BUILDSTDERR:     return self.visit(binop)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 441, in visit
BUILDSTDERR:     return visitor(node, **kwargs)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 563, in visit_BinOp
BUILDSTDERR:     op, op_class, left, right = self._maybe_transform_eq_ne(node)
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 484, in _maybe_transform_eq_ne
BUILDSTDERR:     right = self.visit(node.right, side="right")
BUILDSTDERR:   File "/usr/lib/python3.8/site-packages/pandas/core/computation/expr.py", line 440, in visit
BUILDSTDERR:     visitor = getattr(self, method)
BUILDSTDERR: AttributeError: 'PandasExprVisitor' object has no attribute 'visit_Constant'
BUILDSTDERR: -------------------- >> begin captured logging << --------------------
BUILDSTDERR: numexpr.utils: INFO: Note: NumExpr detected 48 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
BUILDSTDERR: numexpr.utils: INFO: NumExpr defaulting to 8 threads.
BUILDSTDERR: --------------------- >> end captured logging << ---------------------
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Ran 490 tests in 27.429s
BUILDSTDERR: FAILED (errors=6, skipped=41)
BUILDSTDERR: Test failed: <unittest.runner.TextTestResult run=490 errors=6 failures=0>
BUILDSTDERR: error: Test failed: <unittest.runner.TextTestResult run=490 errors=6 failures=0>
```

**Comment by eclipseo** [comment]
Deprecated since version 3.3, will be removed in version 3.8: The behaviour of this function depends on the platform: use perf_counter() or process_time() instead, depending on your requirements, to have a well defined behaviour.
https://docs.python.org/3/library/time.html#time.clock

**Comment by eclipseo** [comment]
'PandasExprVisitor' object has no attribute 'visit_Constant'
https://github.com/pandas-dev/pandas/issues/27261
https://github.com/pandas-dev/pandas/pull/28101

**Comment by QuLogic** [comment]
Ah, thanks for opening this @eclipseo.

**Comment by ljwolf** [comment]
Great, thanks for this report!

- 2 of these test errors (`AttributeError: 'GeometryArray' object has no attribute 'item'`) look like they're due to [the `geopandas` refactoring.](https://jorisvandenbossche.github.io/blog/2019/08/13/geopandas-extension-array-refactor/) Should be simple for us to fix. 
- Test errors in `rtree` look like they're due to the `time.clock` deprecation, and should probably move to the `timeit` module. 
- Will have to investigate the `PandasExprVisitor` failure... not sure I've ever seen that before in the `query()` methods.

**Comment by ljwolf** [comment]
ah yes of course, thank you @eclipseo for the refs on the `PandasExprVisitor` errors.

**Comment by QuLogic** [comment]
3.8 final is out now; any progress on this?

**Comment by QuLogic** [comment]
Most of these are fixed except the `clock` one; opened #240 for that.

**Comment by QuLogic** [comment]
There is one other test failing on Python 3.8: `libpysal.weights.tests.test_adjlist.Test_Adjlist.test_filter`. That is because this set:
https://github.com/pysal/libpysal/blob/facd17c3e45daac34bf51bc186c8710c415865d9/libpysal/weights/adjtools.py#L183
is unordered, and in a different order on 3.8, apparently. The subsequent loop iterates in a different order and drops different elements. A possible workaround might be to do `for tupe in sorted(directed):` here:
https://github.com/pysal/libpysal/blob/facd17c3e45daac34bf51bc186c8710c415865d9/libpysal/weights/adjtools.py#L185

**Comment by QuLogic** [comment]
That last one should be fixed by #244.
